misc/python/materialize/checks/owners.py

```diff
@@ -0,0 +1,82 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.util import MzVersion
+
+
+class Owners(Check):
+    def _can_run(self) -> bool:
+        return self.base_version >= MzVersion.parse("0.47.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive("")
```

Comment:
The general philosophy of the platform checks framework is that you create a dependency tree between objects and then make sure you create/populate the objects from the distinct stages of the dependency graph at each of the phases. This ensures that objects created in a prior phase (that is, in a prior Mz version or prior to some Mz restart, or in the future backup/restore) can still be used to create further objects in subsequent phases.

In this particular case, the dependency graph seems to be like this:
1. CREATE ROLE
2. login as role + CREATE some object
3. CREATE another object that references first object, perform DDL on first object, perform DML on first object, etc.

If none of the items in #3A are currently on the table, then the dependency graph is much simpler , just items 1 and 2 . Those two items then should be happening in different phases of the test.

Therefore, to overly-exercise it, I would suggest something like this:

* initialize:
- create role r1
* manipulate phase 1
- login as r1 , create object as **r1**
- create role r2
* manipulate phase 2
- login as r1, create object as r1
- login as r2, create object as **r2**
- create role r3, login as r3, create object as r3
* validate (can run several times, so must be idempotent)
- validate ownership of each of the previously-created objects
- login as each prior role, create object, check its ownership, drop it

I think you have most of that stuff already, except reusing roles from prior phases in subsequent ones.

There is quite a bit of a potential for a combinatorial explosion here, especially if item No 3 is also going to land in the future, so there is no expecation for complete exhaustiveness.